pkg/plugin/plugin.go

```diff
@@ -110,15 +110,16 @@ func (p *plugin) CreateOrUpdate(ctx context.Context, cs *api.OpenShiftManagedClu
 		err = p.clusterUpgrader.Deploy(ctx, cs, azuretemplate, deployFn)
 	}
 	if err != nil {
-		return err
+		return api.PluginError{Err: err, Step: api.PluginStepDeploy}
```

Comment:
should I return a diff step on deploy vs update?

Reply:
this isn't right.  Update and Deploy already effectively return *PluginError, it's just that their signatures are wrong.  Change their signatures.  Ditto WaitForInfraServices, etc.